libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -153,7 +153,7 @@ internal class RouteRefreshController(
                 }
                 timeUntilNextAttempt = async { delay(routeRefreshOptions.intervalMillis) }
                 val routeRefreshRequestData = routeRefreshRequestDataProvider
-                    .getRouteRefreshRequestDataOrWait()
+                    .getRouteRefreshRequestDataOrWait(routes.first().routeOptions)
```

Comment:
I think this should be moved down to `refreshRoutesOrNull` and we should fetch the options specific to the route we're refreshing since each route can have different options. This is a little counter-intuitive but using this fallback in general is counter intuitive because the car will have a different state than during the initial request. All in all, imo moving it down is the correct way to do it.

Reply:
I thought so as well but the updates are common for all routes (`MapboxNavigation#onEVDataUpdated`). So it's not really specific to one route. Now I'm thinking that maybe not `MapboxNavigation` but `NavigationRoute` should receive updates. But I'm not sure it would be very convenient for end users.
Also look at line 172: which route should I use there? We fix the request data for all routes. Because we might have had different leg index for different routes if we fixed the data for one specific route.

Reply:
> updates are common for all routes (MapboxNavigation#onEVDataUpdated). So it's not really specific to one route. Now I'm thinking that maybe not MapboxNavigation but NavigationRoute should receive updates.

No, I think the current pattern is correct. We have only one vehicle so its parameters should be used to refresh all routes.

What we're considering here is just a fallback but if we use the route options as the source of data, we should be using the one that's relevant for the route we're refreshing. As I mentioned it is all counter intuitive for EV data which generally should be provided externally and for all routes but I wanted to avoid introducing a pattern for the future where actually using data specific to the route has bigger consequences. The worst edge case I can see is if the first route was not an EV route and the others were, then the refresh for the EV routes would fail. This is more theoretical than realistic but it's the type of inconsistency that we're introducing.

The current PR solves majority of the existing realistic problems so I'm fine merging it and we can iterate further.

> Also look at line 172: which route should I use there?

We'd have to refactor `RefreshedRouteInfo` (and probably `RouteRefreshRequestDataProvider`), it doesn't need to hold the whole `RouteRefreshRequestData`, it only needs the leg index.

In any case, let's follow up separately.

Reply:
#6543